### PR TITLE
timeout after reading HTTP request for 10 minutes

### PR DESCRIPTION
**Pull request by GregorioDiStefano**

### CODE REVIEW

**Comment by GregorioDiStefano** [comment]
let me check if this fixes the issue...

**Comment by GregorioDiStefano** [comment]
this fixes the timeout issue.

**Comment by GregorioDiStefano** [comment]
@bboozzoo

**Comment by kjaskiewiczz** [comment]
is it workaround for creating deployment with many devices problem?

**Comment by GregorioDiStefano** [comment]
@kjaskiewiczz no, when I tried uploading an image of 2GB to mender running on AWS, the upload times out.

**Comment by kjaskiewiczz** [comment]
We've got hardcoded limit for artifact file size - 10GB.
Will 600s be enough for uploading ~10GB file?
Or we are just looking for something good enough for files with size ~2GB?

**Comment by GregorioDiStefano** [comment]
I'm looking for something good enough for 2GB, because Eystein states thats the type rootfs size. 
Maybe I should increase it to 20-30 minutes?

File: nginx.conf [hunk 1]
```diff
@@ -43,6 +43,8 @@ http {
         ssl_stapling on;
         ssl_stapling_verify on;
 
+        proxy_read_timeout 600;
```
Comment (bboozzoo) [inline nginx.conf]: For sure this should be limited to `/api/integrations/0.1/deployments/images` endpoint only.

**Comment by bboozzoo** [comment]
IMO a browser will timeout sooner than 10 minutes. 

One thing, the API design might be incorrect for this specific case. I think that a sane thing to do here, would be to `202 Accepted` and just expose an endpoint where one can monitor upload progress. Right now the AWS upload is piggybacked on the actual request, so whatever AWS upload delay is, it becomes part of client request processing delay. @maciejmrowiec @marcin @mendersoftware/rndity thoughts?

**Comment by GregorioDiStefano** [comment]
@bboozzoo I increased the timeout to 30 minutes

**Pull request closed without merging**